### PR TITLE
add https support

### PR DESCRIPTION
**Pull request by lk29**
add https support

more config file options
more command line options
more logs

### CODE REVIEW

**Comment by 117503445** [comment]
Thank you very much for the PR!

My personal habit is to provide https functionality through a proxy such as traefik. Because certificate management is a complex thing that requires regular rotation and possibly access to let's encrypt and acme for automated certificate requests.

I've been busy lately, so I'll take a closer look at the code when I have time. Thanks again for your contribution!

**Comment by lk29** [comment]
use openssl for selfsigned
`openssl req -x509 -newkey rsa:4096 -keyout localhost.key -out localhost.crt -sha256 -days 3650

if you have access to dns management
To obtain a certificate, you can use authorization via dns
`certbot -d domainname  --manual --preferred-challenges dns certonly

it is much better and safer than using http

**Comment by 117503445** [comment]
I'm sorry, but I don't want this PR to be merged. The main reason is that I want to provide HTTPS through a gateway rather than GoWebDAV itself. On the one hand, a gateway like Traefik supports automatic Let's Encrypt certificate acquisition through ACME, which is difficult for GoWebDAV to achieve. On the other hand, if GoWebDAV occupies port 443, other applications cannot use port 443, but a gateway can forward requests to different application ports based on the host, thereby achieving reuse.

In summary, I do not want to increase the complexity of the GoWebDAV configuration for this feature. But thank you for your contribution, and those who need it are welcome to use your branch :)
